src/Tribe/Views/V2/Template_Bootstrap.php

```diff
@@ -323,15 +324,38 @@ public function should_load( $query = null ) {
 		 *
 		 * @see \Tribe__Events__Query::parse_query() where this property is set.
 		 */
-		$should_load = $query->is_main_query() && ! empty( $query->tribe_is_event_query );
+		$should_load = $query->is_main_query() && ! empty( $query->tribe_is_event_query ); // here
+
+		if ( ! $should_load ) {
+			/**
+			 * Sometimes, tribe_is_event_query has not been set when we call this function.
+			 * Typically for shortcodes, and usually when called by a page builder
+			 * OR the query has been filtered by another plugin.
+			 *
+			 * So we test a different way when we can.
+			 */
+			$is_shortcode   = false;
+			if ( $query->is_page ) {
+				$queried_object = $query->queried_object;
+
+				if ( ! empty( $queried_object->post_content ) ) {
+					// Classic Editor
+					$is_shortcode = has_shortcode( 'tribe_events', $queried_object->post_content );
+					// Bollocks
+					$is_shortcode = $is_shortcode || false !== stripos( $queried_object->post_content, '[tribe_events');
+				}
+			}
+
+			$should_load = $should_load || $is_shortcode;
+		}
```

Comment:
I see what this does, but I am not entirely sure why this is needed for shortcodes in specific.

I would love to dig a bit more into why we are doing it, instead of the actual fix which is valid, but not super "clean".